azure-migrate/dependencies-at-scale/AzMig_Dependencies.ps1

```diff
@@ -0,0 +1,585 @@
+}
+
+
+function Get-AzMigProject {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory = $true)][string]$ResourceGroupName,
+        [Parameter(Mandatory = $true)][string]$ProjectName
+    )
+
+    $query = "resources | where type == 'microsoft.migrate/migrateprojects' and resourceGroup == '$ResourceGroupName' and name == '$ProjectName'"
+    $result = $null
+    $result = Search-AzGraph -Query $query
+    if (-not $result) {throw "Project not found"}
+    return $result.Id
+
+}
+
+
+function Get-AzMigDiscoveredVMwareVMs {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory = $true)][string]$ResourceGroupName,
+        [Parameter(Mandatory = $true)][string]$ProjectName,
+        [Parameter(Mandatory = $true)][string]$OutputCsvFile = "VMwareVMs.csv",
+	    [Parameter()][string]$ApplianceName = $null
+    )
+
+    if(-not (Test-Path -IsValid -Path $OutputCsvFile)) {
+        throw "Output CSV file path is not valid"
+    }
+
+    if (Test-path -Path $OutputCsvFile) {
+        throw "File $OutputCsvFile already exists. Specify a different name or path"
+    }
+	
+	if (-not ($OutputCsvFile -match ".*\.csv$")) {
+        throw "Output file specified is not CSV."    
+    }
+    
+    $ProjectId = Get-AzMigProject -ResourceGroupName $ResourceGroupName -ProjectName $ProjectName
+	
+       
+    $Properties = GetRequestProperties
+
+    if(-not $ProjectId) {throw "Project ID is invalid"};
+    #GetApplianceToSiteMap
+    $query = "resources| where type == 'microsoft.offazure/vmwaresites' or type == 'microsoft.offazure/hypervsites' or type == 'microsoft.offazure/serversites'| where resourceGroup == '$ResourceGroupName' and properties.discoverySolutionId has '$ProjectName'| project properties.applianceName, id"
+    $response = $null
+    $response = Search-AzGraph -Query $query
+    if (-not $response) {throw "appliance not found"}
+    $appMap = @{}
+
+    # Iterate over the query result
+    foreach($row in $response){    
+    $applianceName = $row.properties_applianceName
+    $id = $row.id
+  
+    # Map the 'applianceName' to the 'id'
+    $appMap[$applianceName] = $id
+    }
+
+    $vmwareappliancemap = @{}
+
+    #Discard non-VMware appliances
+    #If Appliance name is passed get data only for that appliance
+    #If Appliance name is not passed , get data for all appliances in that project
+    if (-not $ApplianceName){
+	$appMap.GetEnumerator() | foreach {if($_.Value -match "VMwareSites|HyperVSites|ServerSites") {$vmwareappliancemap[$_.Key] = $_.Value}}}else{
+	$appMap.GetEnumerator() | foreach {if($_.Value -match "VMwareSites|HyperVSites|ServerSites" -and $_.Key -eq $ApplianceName) {$vmwareappliancemap[$_.Key] = $_.Value}}}
+    Write-Debug $vmwareappliancemap.count
+    if($vmwareappliancemap) {$vmwareappliancemap | Out-String | Write-Debug};
+    if (-not $vmwareappliancemap.count) {throw "No VMware VMs discovered in project"};
+    
+	Write-Host "Please wait while the list of discovered machines is downloaded..."
```

Comment:
Keep in mind that `Write-Host` doesn't write to the pipeline and this comment won't be logged or seen by anyone if the function is being used in an automation scenario.